internal/common/resource.go

```diff
@@ -431,3 +432,108 @@ func ResourceDeletedResult(resource client.Object, res OperationResult) Reconcil
 		OperationResult: res,
 	}
 }
+
+func defaultUpdateFunc(newObj, foundObj client.Object) {
+	switch newTyped := newObj.(type) {
+	case *core.ConfigMap:
+		foundConfigMap := foundObj.(*core.ConfigMap)
+		foundConfigMap.Immutable = newTyped.Immutable
+		foundConfigMap.Data = newTyped.Data
+		foundConfigMap.BinaryData = newTyped.BinaryData
+
+	case *core.Namespace:
+		// Intentionally empty
+
+	case *core.Service:
+		foundService := foundObj.(*core.Service)
+		// ClusterIP should not be updated
+		newTyped.Spec.ClusterIP = foundService.Spec.ClusterIP
+		foundService.Spec = newTyped.Spec
+
+	case *core.ServiceAccount:
+		// Intentionally empty
+
+	case *rbac.ClusterRole:
+		foundObj.(*rbac.ClusterRole).Rules = newTyped.Rules
+
+	case *rbac.ClusterRoleBinding:
+		foundBinding := foundObj.(*rbac.ClusterRoleBinding)
+		foundBinding.RoleRef = newTyped.RoleRef
+		foundBinding.Subjects = newTyped.Subjects
+
+	case *rbac.Role:
+		foundObj.(*rbac.Role).Rules = newTyped.Rules
+
+	case *rbac.RoleBinding:
+		foundBinding := foundObj.(*rbac.RoleBinding)
+		foundBinding.RoleRef = newTyped.RoleRef
+		foundBinding.Subjects = newTyped.Subjects
+
+	case *apps.DaemonSet:
+		foundObj.(*apps.DaemonSet).Spec = newTyped.Spec
+
+	case *apps.Deployment:
+		foundObj.(*apps.Deployment).Spec = newTyped.Spec
+
+	case *instancetypev1alpha2.VirtualMachineClusterInstancetype:
+		foundObj.(*instancetypev1alpha2.VirtualMachineClusterInstancetype).Spec = newTyped.Spec
+
+	case *instancetypev1alpha2.VirtualMachineClusterPreference:
+		foundObj.(*instancetypev1alpha2.VirtualMachineClusterPreference).Spec = newTyped.Spec
+
+	case *routev1.Route:
+		foundObj.(*routev1.Route).Spec = newTyped.Spec
+
+	case *promv1.PrometheusRule:
+		foundObj.(*promv1.PrometheusRule).Spec = newTyped.Spec
+
+	case *promv1.ServiceMonitor:
+		foundObj.(*promv1.ServiceMonitor).Spec = newTyped.Spec
+
+	default:
+		panic(fmt.Sprintf("Default update is not supported for type: %T", newObj))
```

Comment:
We'll need to add also `pipeline.Task` and `pipeline.Pipeline` after your PR is merged.